### PR TITLE
[DROOLS-3150] Avoid multiple "editing" cell in header

### PR DESCRIPTION
**Pull request by gitgabrio**
@kkufova @Rikkola @danielezonca 

Scope of this PR is to fix 

https://issues.jboss.org/browse/DROOLS-3150

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/metadata/ScenarioHeaderMetaData.java [hunk 1]
```diff
@@ -76,9 +77,17 @@ public void edit(final GridBodyCellEditContext context) {
         if (readOnly) {
             throw new IllegalStateException("A read only header cannot be edited");
         }
-        factory.attachDomElement(context,
-                                 (e) -> e.getWidget().setText(getTitle()),
-                                 (e) -> e.getWidget().setFocus(true));
+        if (!isEditing) {
```
Comment (danielezonca) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/metadata/ScenarioHeaderMetaData.java]: Can you please update/add a test to cover this new branch?

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/widgets/ScenarioGridCell.java [hunk 2]
```diff
@@ -20,7 +20,9 @@
 
 public class ScenarioGridCell extends BaseGridCell<String> {
 
+    // true if this cell is already in editing mode
     private boolean isEditing = false;
+    // true if this cell has to show errors
     private boolean isError = false;
```
Comment (Rikkola) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/widgets/ScenarioGridCell.java]: If you rename for example to `isInEditingMode `mode and to `isHavingAnError ` the comments can be left out.

**Comment by gitgabrio** [comment]
@Rikkola @danielezonca 
Add tests

**Comment by gitgabrio** [comment]
@manstis 
Could you please merge this, squashing commits? Thanks!

**Comment by manstis** [comment]
@gitgabrio Please let me know when it has a green build. Thanks.

**Pull request merged**